### PR TITLE
Enable tape mode in tests

### PR DESCRIPTION
**Pull request by mariaschuld**
**Context:** 
To get the new tape core - which is at the moment an experimental feature sitting side-by-side to the former core - to feature parity, we need a way to run integration tests automatically with both cores. Once all such tests pass in tape mode, feature parity is established. 

**Description of the Change:**
Add a fixture to `conftest.py` that runs a test once with tape mode enabled (making sure it is disabled after the test has finished or failed), and once without tape mode enabled.

**Usage:**
Add the fixture to a test or class using ``@pytest.mark.usefixtures("tape_mode")``.

### CODE REVIEW

File: pennylane/numpy/wrapper.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 This module provides the PennyLane wrapper functions for modifying NumPy,
 such that it accepts the PennyLane :class:`~.tensor` class.
 """
-from collections import Sequence
+from collections.abc import Sequence
```
Comment (mariaschuld) [inline pennylane/numpy/wrapper.py:18]: Smuggled this in from seeing warnings about the import.

File: tests/templates/test_embeddings.py [hunk 2]
```diff
@@ -173,6 +173,7 @@ def circuit(x=None):
         # No normalization error is raised
         circuit(x=inputs)
 
+
```
Comment (mariaschuld) [inline tests/templates/test_embeddings.py:176]: Smuggled in this line for proper formatting.

File: tests/templates/test_integration.py [hunk 3]
```diff
@@ -303,6 +303,7 @@
            ]
 
 
+@pytest.mark.usefixtures("tape_mode")
```
Comment (mariaschuld) [inline tests/templates/test_integration.py:306]: Example of how to use the fixture: Now all tests in the class below are run twice. At this stage they fail a lot.

Comment (mariaschuld) [inline tests/templates/test_integration.py:306]:
```suggestion
```

**Review by josh146 — APPROVED** [review]

**Comment by trbromley** [comment]
[ch1677]

**Pull request merged**